codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/RubyCodeWriter.java

```diff
@@ -60,6 +61,50 @@ public RubyCodeWriter(String namespace) {
         putFormatter('T', new RubySymbolFormatter());
     }
 
+    /**
+     * The purpose of this method is to populate module blocks before code population for each file.
```

Comment:
Having module blocks generated in advance actually makes it more convenient because contributor can focus solely on writing out the ruby structures via `writer.write(...)`, and not having handle module blocks.

```
writer
    .includePreamble()
    .includeRequires()
    .openBlock("module SomeService")
    .openBlock("module Types")
    .write("$L", rubyStructures)
    .closeBlock("end")
    .closeBlock("end");
```

I am okay to remove it if you think this is an overkill. But I do think the `ModuleBlockInterceptor` is a nice feature to have.

Reply:
@alextwoods Thoughts?

Reply:
I'm not sure I understand the usage differences.  Can you provide a quick example of usage?

Reply:
Sure, makes sense.

I have removed the modules pre-population so it's up to the contributor to decide on how to write module blocks. The following examples behave similarly:

```
writer
    .openBlock("module SomeService")
    .openBlock("module Types")
    .write("$L", rubyStructures)
    .closeBlock("end")
    .closeBlock("end");
```

```
writer
    .openModule("SomeService")
    .openModule("Types")
    .write("$L", rubyStructures)
    .closeAllModules();
```

> What do other Smithy SDKs do? Is this a common pattern there?

The only other codegen I have looked at is the typescript codegen, and it doesn't generate anything similar to ruby `modules` and mostly uses `openBlock`.

Reply:
Nice - this looks good.